docs/using-the-compiler.rst

```diff
@@ -0,0 +1,141 @@
+files reside, so things like ``import "/etc/passwd";`` only work if you add ``=/`` as a remapping.
+
+If there are multiple matches due to remappings, the one with the longest common prefix is selected.
+
+If your contracts use :ref:`libraries <libraries>`, you will notice that the bytecode contains substrings of the form ``__LibraryName______``. You can use ``solc`` as a linker meaning that it will insert the library addresses for you at those points:
+
+Either add ``--libraries "Math:0x12345678901234567890 Heap:0xabcdef0123456"`` to your command to provide an address for each library or store the string in a file (one library per line) and run ``solc`` using ``--libraries fileName``.
+
+If ``solc`` is called with the option ``--link``, all input files are interpreted to be unlinked binaries (hex-encoded) in the ``__LibraryName____``-format given above and are linked in-place (if the input is read from stdin, it is written to stdout). All options except ``--libraries`` are ignored (including ``-o``) in this case.
+
+
+**************************************************
+Standardized Input Description and Metadata Output
+**************************************************
+
+In order to ease source code verification of complex contracts that are spread across several files,
+there is a standardized for describing the relations between those files.
+Furthermore, the compiler can generate a json file while compiling that includes
+the source, natspec comments and other metadata whose hash is included in the
+actual bytecode.
+
+There is some overlap between the input description and the metadata output
+and due to the fact that some fields are optional, the metadata can be used as
+input to the compiler. In order to verify the metadata, you actually take it,
+re-run the compiler on the metadata and check that it again produces the same
+metadata.
+
+If the compiler is invoked in a different way, not using the input
+description (for example by using a file content retrieval callback),
+the compiler can still generate the metadata alongside the bytecode of each
+contract.
+
+The metadata standard is versioned. Future versions are only required to provide the "version" field,
+the two keys inside the "compiler" field. The field compiler.keccak should be the keccak hash
+of a binary of the compiler with the given version.
+
+The example below is presented in a human-readable way. Properly formatted metadata
+should use quotes correctly, reduce whitespace to a minimum and sort the keys of all objects
+to arrive at a unique formatting.
+
+Comments are of course not permitted and used here only for explanatory purposes.
+
+Input Description
+-----------------
+
+The input description could change with each compiler version, but it
+should be backwards compatible if possible.
+
+    {
+      sources:
+      {
+        "abc": "contract b{}",
+        "def": {keccak: "0x123..."}, // source has to be retrieved by its hash
```

Comment:
As mentioned in the Gist, I think this should have a (optional) `type`/`kind`/`network` field to signal where to look for it, i.e. `"swarm"`, `"ipfs"`.